### PR TITLE
OCD-3695: Report deprecated endpoint usage by API Key

### PR DESCRIPTION
**Pull request by kekey1**

### CODE REVIEW

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/filter/DeprecatedEndpointUsageFilter.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                        deprecatedApiUsageDao.createOrUpdateDeprecatedApiUsage(deprecatedApiUsage);
+                    }
+                } else {
+                    LOGGER.error("Could not determine unique matching URL Pattern for " + request.getMethod()
+                        + " Request: " + request.getRequestURI());
+                }
+            }
+        }
+
+        //TODO: something about deprecated parameters
```
Comment (andlar) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/filter/DeprecatedEndpointUsageFilter.java]: Todo in a later ticket? Or something for now?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/api/domain/ApiKey.java [hunk 2]
```diff
@@ -2,54 +2,50 @@
 
 import java.io.Serializable;
 import java.util.Date;
+import java.util.Objects;
 
-import gov.healthit.chpl.util.Util;
+import javax.xml.bind.annotation.XmlTransient;
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/api/domain/ApiKey.java]: Do these API-Keys show up in XML anywhere? I wouldn't have guessed that they would, which makes me wonder why they need to be marked with XmlTransient

**Review by tmy1313 — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/api/domain/ApiKey.java (hunk 2 above)
Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/api/domain/ApiKey.java]: No they don't. I just copied it from somewhere else. Removed.

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/filter/DeprecatedEndpointUsageFilter.java (hunk 1 above)
Comment (kekey1) [reply chpl/chpl-api/src/main/java/gov/healthit/chpl/filter/DeprecatedEndpointUsageFilter.java]: Later ticket. Removed

**Review by andlar — DISMISSED** [review]

**Review by tmy1313 — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by tmy1313 — APPROVED** [review]

**Pull request merged**